### PR TITLE
FEATURE: Add sections to personal schedule directly from search

### LINKED ISSUE #7: FEATURE: Add sections to personal schedule directly from search
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

add a course/section from the search page to a personal schedule

### So that...

more easily add courses to my schedule

### Discussion

Currently, adding a course to a personal schedule requires remembering/copy pasting the enroll code. It would be much nicer for UX if you can create add courses from searches directly.

### Acceptance Criteria

- [x] #15
- [ ] Clicking this button opens a modal, to select an existing personal schedule to add this course to.

- [x] The schedule selection should be by _name_, not by id.
- [ ] The "add" button only appears for sections, not courses (since currently only individual sections, not courses can be added).
- [ ] *Bonus*: In the modal, there is a link to to the "add new personal schedule" page, so user can create one from there if they have not already.

### Implementation Todos

Frontend:
- [x] Create a react component that allows selecting from existing personal schedules by _name_, not by id. Note: the ID may be used internally in the code, just not visible to the user.
- [x] Create a modal/webpage for adding a new course. This should use the new schedule selection component.
- [ ] In search results table, add an "add" button to applicable section that open the modal.

Backend:
- It _may_ be possible to implement the frontend without adding new backend endpoints. However, additional endpoints can be added if necessary.

Testing:
- [ ] Front end tests pass and there is adequate coverage

### PR DESCRIPTION
**Pull request by theanimated01**
Closes #7 

Dev deployment: https://proj-courses-theanimated01-dev.dokku-07.cs.ucsb.edu/

This PR adds a feature where you can directly add courses to a schedule from the courses search page.

Changes made:

1. Edited SectionTable.js to include a new modal when you press the Add Button
2. Edited SectionTableBase.js to include a column with the Add button. Add button only appears beside sections when a course is expanded. If a course does not have sections (i.e. not expandable) the Add button will appear beside the course
3. Modified the useBackend.js under utils so I could call that hook based on a boolean value. It was required otherwise the homepage would throw a error 403 (communicating with backend) as we are not signed in when we first open the home page and I used the backend call to populate the schedule dropdown in the modal. The boolean allowed me to call the useBackend function only when the modal was opened.
4. Edited test file for SectionTable

Screenshots for before and after:
Screenshot 2024-03-07 at 12 22 05 AM
Screenshot 2024-03-07 at 5 28 42 PM
Screenshot 2024-03-07 at 5 30 29 PM

### CODE REVIEW

**Review by bwr02 — DISMISSED** [review]
LGTM. Amazing guys great work

**Review by chloeqta — DISMISSED** [review]
LGTM

**Review by AnikaArora — CHANGES_REQUESTED** [review]
Overall, looks great! One small issue is that when you're adding a class that doesn't have a section, the "add" button doesn't look like a button because its background really blends into the cell's color. Could you please edit the UI to made it look more like a button? When the class has a section, it's very clear since there's a color contrast but here it's very blended. 

Also, please add a screenshot of the "after" for a class that doesn't have a section too (once you fix this UI thing)!

**Review by bwr02 — DISMISSED** [review]
LGTM

File: frontend/src/main/components/SectionsTableBase.js [hunk 1]
```diff
@@ -55,10 +78,10 @@ export default function SectionsTableBase({
                         // Stryker disable next-line ObjectLiteral
                         style={{
                           background: cell.isGrouped
-                            ? "#34859b"
+                            ? "#7baacb"
```
Comment (tallyhawley) [inline frontend/src/main/components/SectionsTableBase.js]: Is it possible to change the color of the button instead of changing the table colors?

Comment (theanimated01) [reply frontend/src/main/components/SectionsTableBase.js]: Done. changed color of button instead of table

**Review by sophiattran — APPROVED** [review]
lgtm!

**Review by bwr02 — APPROVED** [review]
LGTM

**Review by AnikaArora — APPROVED** [review]
LGTM!

**Pull request merged**